tests/integration/main.go

```diff
@@ -24,46 +26,63 @@ var localSdkPath = flag.String("local-sdk-path", "",
 func main() {
 	flag.Parse()
 
-	// wg waits for all runners to complete
-	var wg sync.WaitGroup
-
-	start := time.Now()
-	tr := DefaultTestRun()
-	tr.SetLocalSDKPath(*localSdkPath)
-	tr.ValidateStringLiterals()
-	tr.startDocker()
-
-	dmc := DemocracyTestRun()
-	dmc.SetLocalSDKPath(*localSdkPath)
-	dmc.ValidateStringLiterals()
-	dmc.startDocker()
-
-	slash := SlashThrottleTestRun()
-	slash.SetLocalSDKPath(*localSdkPath)
-	slash.ValidateStringLiterals()
-	slash.startDocker()
-
-	if *multiconsumer {
-		mul := MultiConsumerTestRun()
-		mul.SetLocalSDKPath(*localSdkPath)
-		mul.ValidateStringLiterals()
-		mul.startDocker()
+	if happyPathOnly != nil && *happyPathOnly {
+		fmt.Println("=============== running happy path only ===============")
+		tr := DefaultTestRun()
+		tr.SetLocalSDKPath(*localSdkPath)
+		tr.ValidateStringLiterals()
+		tr.startDocker()
+		tr.ExecuteSteps(happyPathSteps)
+		tr.teardownDocker()
+		return
+	}
 
-		wg.Add(1)
-		go mul.ExecuteSteps(&wg, multipleConsumers)
+	testRuns := []testRunWithSteps{
+		{DefaultTestRun(), happyPathSteps},
+		{DemocracyTestRun(), democracySteps},
+		{SlashThrottleTestRun(), slashThrottleSteps},
+	}
+	if includeMultiConsumer != nil && *includeMultiConsumer {
+		testRuns = append(testRuns, testRunWithSteps{MultiConsumerTestRun(), multipleConsumers})
 	}
 
-	wg.Add(1)
-	go tr.ExecuteSteps(&wg, happyPathSteps)
+	start := time.Now()
+	if parallel != nil && *parallel {
+		fmt.Println("=============== running all tests in parallel ===============")
+		var wg sync.WaitGroup
+		for _, run := range testRuns {
+			wg.Add(1)
+			go func(run testRunWithSteps) {
+				defer wg.Done()
+				tr := run.testRun
+				tr.Run(run.steps, *localSdkPath)
+			}(run)
+		}
+		wg.Wait()
+		fmt.Printf("TOTAL TIME ELAPSED: %v\n", time.Since(start))
+		return
+	}
 
-	wg.Add(1)
-	go dmc.ExecuteSteps(&wg, democracySteps)
+	for _, run := range testRuns {
+		tr := run.testRun
+		tr.Run(run.steps, *localSdkPath)
+	}
+	fmt.Printf("TOTAL TIME ELAPSED: %v\n", time.Since(start))
+}
 
-	wg.Add(1)
-	go slash.ExecuteSteps(&wg, slashThrottleSteps)
+// Run sets up docker container and executs the steps in the test run.
+// Docker containers are torn down after the test run is complete.
+func (tr *TestRun) Run(steps []Step, localSdkPath string) {
```

Comment:
nit: it would be nice to have all  the`TestRun` methods unexported for consistency.

Reply:
Done